gnocchi/incoming/redis.py

```diff
@@ -93,6 +96,10 @@ def add_measures_batch(self, metrics_and_measures):
                 notified_sacks.add(sack_name)
         pipe.execute()
 
+    @tenacity.retry(
+        wait=utils.wait_exponential,
+        # Never retry except when explicitly asked by raising TryAgain
+        retry=tenacity.retry_never)
```

Comment:
I'd go with:
```suggestion
    @tenacity.retry(
        wait=utils.wait_exponential,
        retry=tenacity.retry_if_exception_type(ConnectionError))
```
That should be simpler.
You probably want to stop at some point too though, I'd add a condition for it to stop retrying after a few tries.

Reply:
That looks better. Thank you for the suggestion.

I've never seen this repeated more than twice for now.